tree-sitter/src/TreeSitter/GenerateSyntax.hs

```diff
@@ -82,14 +92,13 @@ syntaxDatatype language datatype = skipDefined $ do
 
 
 -- | Create TH-generated SymbolMatching instances for sums, products, leaves
-symbolMatchingInstance :: Ptr TS.Language -> Name -> String -> Q [Dec]
-symbolMatchingInstance language name str = do
-  tsSymbol <- runIO $ withCString str (TS.ts_language_symbol_for_name language)
-  tsSymbolType <- toEnum <$> runIO (TS.ts_language_symbol_type language tsSymbol)
+symbolMatchingInstance :: [(String, Named)] -> Name -> Named -> String -> Q [Dec]
+symbolMatchingInstance allSymbols name named str = do
+  let tsSymbols = elemIndices (str, named) allSymbols
```

Comment:
Since we're going to be calling `symbolMatch` a lot, I wonder if there's some other data structure that we should be using for `tsSymbols` instead of a list, like a `Set` or something, which might have a faster `elem`.

Usually these lists will only have 1 element, and only occasionally will they have two or three, so maybe a list is fine. Just wanted to raise this in case anyone more familiar w/ haskell data structures has a preference here.

Reply:
tbqh I'm not sure either, maybe @patrickt can chime in?